src/com/google/javascript/jscomp/PolymerClassRewriter.java

```diff
@@ -254,8 +258,36 @@ void rewritePolymerClassDeclaration(
     // If property renaming is enabled, wrap the properties object literal
     // in a reflection call so that the properties are renamed consistently
     // with the class members.
+    //
+    // Also add reflection and sinks for computed properties and complex observers
+    // and switch simple observers to direct function references.
     if (propertyRenamingEnabled && cls.descriptor != null) {
-      addObjectReflectionCall(cls, cls.descriptor);
+      convertSimpleObserverStringsToReferences(cls);
+      List<Node> propertySinks = new ArrayList<>();
+      if (polymerExportPolicy != PolymerExportPolicy.EXPORT_ALL) {
+        propertySinks.addAll(addComputedPropertiesReflectionCalls(cls));
+        propertySinks.addAll(addComplexObserverReflectionCalls(cls));
+      }
+
+      if (propertySinks.size() > 0) {
+        if (traversal.getScope().getVar(CheckSideEffects.PROTECTOR_FN) == null) {
```

Comment:
This check might be what's causing the crash. NodeTraversal doesn't automatically regenerate parent scopes when they change, so if this is called twice while in the same traversal, it will add the extern both times.

Reply:
Oh interesting - I had no idea.

I pushed a property check up to keep track and hopefully ensure it is only injected once.